src/elf.rs

```diff
@@ -905,6 +960,23 @@ impl<E: UserDefinedError, I: InstructionMeter> Executable<E, I> {
     }
 }
 
+pub(crate) fn get_ro_region(ro_section: &Section, elf: &[u8]) -> MemoryRegion {
+    let (offset, ro_data) = match ro_section {
+        Section::Owned(s) => (0, s.as_slice()),
+        Section::Borrowed(range) => (range.start, &elf[range.clone()]),
+    };
+
+    // If offset > 0, the region will start at MM_PROGRAM_START + the offset of
```

Comment:
This should be the only slightly controversial change. Given the following ELF layout:

```
[leading data][ro data][trailing data]
```

- Accessing `[trailing data]` (1 byte past the last readonly byte), always returns `EbpfError::InvalidVirtualAddress`.
- When there's gaps between the sections that form `[ro data]`, the data is copied and the gaps, including `[trailing data]` are filled with zeroes.
- When there's no gaps between the sections  that form `[ro data]`, the data is borrowed and accessing `[leading data]` returns `EbpfError::InvalidVirtualAddress`. This is so that we don't expose bits of the input ELF that are not part of rodata. 

So to sum up the difference: owned sections fill the gap until the beginnig of rodata with zeros; when borrowing we can't fill, so we return an error.

I'm now going to add a config switch to toggle borrowed ro sections and my question is: if the switch is on, should I also go ahead and return `EbpfError::InvalidVirtualAddress` when accessing `[trailing data]` with owned sections too?

Reply:
That sounds consistent and also throws a legit error because the program should not be accessing elf data outside of the text or ro sections.

Reply:
Ok this is now implemented behind `Config::optimize_rodata`.